apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.java

```diff
@@ -25,10 +27,22 @@
 
   @Nonnull ApolloCall<T> clone();
 
-  interface Callback<T> {
-    void onResponse(@Nonnull Response<T> response);
+  abstract class Callback<T> {
+    public abstract void onResponse(@Nonnull Response<T> response);
+
+    public abstract void onFailure(@Nonnull ApolloException e);
+
+    public void onHttpError(@Nonnull ApolloHttpException e) {
+      onFailure(e);
+    }
+
+    public void onNetworkError(@Nonnull ApolloNetworkException e) {
```

Comment:
Should we have the exception `ApolloNetworkException`? I think an `IOException` is more fitting for this case